### PR TITLE
Make docs build warnings into errors

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

See https://github.com/NREL/OpenStudio-HPXML/pull/1131 and https://github.com/NREL/resstock/pull/950 for context.

## Checklist

Not all may apply

- [ ] Code changes (must work)
- [ ] Tests exercising your feature/bug fix (check coverage report on Checks -> BuildStockBatch Tests -> Artifacts)
- [ ] Coverage has increased or at least not decreased. Update `minimum_coverage` in `.github/workflows/ci.yml` as necessary.
- [ ] All other unit tests passing
- [ ] Update validation for project config yaml file changes
- [ ] Update existing documentation
- [ ] Run a small batch run to make sure it all works (local is fine, unless an Eagle specific feature)
- [ ] Add to the changelog_dev.rst file and propose migration text in the pull request

### CODE REVIEW

File: buildstockbatch/workflow_generator/residential_hpxml.py [hunk 1]
```diff
@@ -82,7 +82,16 @@ def create_osw(self, sim_id, building_id, upgrade_idx):
...
+        sim_ctl_args = {
+            'simulation_control_timestep': 60,
+            'simulation_control_run_period_begin_month': 1,
+            'simulation_control_run_period_begin_day_of_month': 1,
+            'simulation_control_run_period_end_month': 12,
+            'simulation_control_run_period_end_day_of_month': 31,
+            'simulation_control_run_period_calendar_year': 2007,
+            'debug': False,
+            'add_component_loads': False
+        }
```
Comment (nmerket) [inline buildstockbatch/workflow_generator/residential_hpxml.py]: This seems out of scope for this PR.

File: docs/conf.py [hunk 2]
```diff
@@ -222,3 +222,19 @@
...
+  ('py:func', 'BuildStockBatchBase.validate_xor_nor_schema_keys'),
+  ('py:func', 'EagleBatch.run_building'),
+  ('py:class', 'sampler.CommercialSobolSingularitySampler'),
+  ('py:class', 'sampler.CommercialSobolDockerSampler'),
+  ('py:class', 'workflow_generator.CommercialDefaultWorkflowGenerator'),
+  ('py:class', 'sampler.PrecomputedSampler'),
+  ('py:class', 'sampler.BuildStockSampler'),
+  ('py:class', 'BuildStockBatchBase'),
+  ('py:func', 'BuildStockBatchBase.run_sampling')
+]
```
Comment (nmerket) [inline docs/conf.py:240]: Any idea why these aren't being found?

File: buildstockbatch/workflow_generator/residential_hpxml.py (hunk 1 above)
Comment (joseph-robertson) [reply buildstockbatch/workflow_generator/residential_hpxml.py]: It _is_, but it _isn't_.

It prevents a warning that the hash is empty, and I'm doing it in this pending PR anyway: https://github.com/NREL/buildstockbatch/pull/274.

File: docs/conf.py (hunk 2 above)
Comment (joseph-robertson) [reply docs/conf.py:240]: I think these are all references in old changelog and migration files, and do not exist anymore. Edit: Oh, maybe it's a mixture of not existing and bad paths?

Comment (joseph-robertson) [reply docs/conf.py:240]: Here is the failed ci that lists them: https://github.com/NREL/buildstockbatch/runs/7046343242?check_suite_focus=true

**Review by nmerket — APPROVED** [review]
